BetaUtils.Core/Exceptions/Middleware/ExceptionMiddleware.cs

```diff
@@ -0,0 +1,77 @@
+﻿using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Abstractions;
+using Microsoft.AspNetCore.Mvc.Infrastructure;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+using Microsoft.AspNetCore.Routing;
+
+namespace BetaUtils.Core.Exceptions.Middleware;
+
+/// <summary>
+/// Handling exceptions that occur during the execution of subsequent middleware components in an ASP.NET Core application
+/// pipeline. It catches exceptions, generates appropriate error responses, and adds them to the HTTP context.
+/// </summary>
+public class ExceptionMiddleware
+{
+    private static readonly ActionDescriptor EmptyActionDescriptor = new();
+    private static readonly RouteData EmptyRouteData = new();
+
+    public ExceptionMiddleware(RequestDelegate next, IActionResultExecutor<ObjectResult> executor)
+    {
+        Next = next;
+        Executor = executor;
+    }
+
+    private RequestDelegate Next { get; }
+
+    private IActionResultExecutor<ObjectResult> Executor { get; }
+
+    public async Task Invoke(HttpContext context)
+    {
+        try
+        {
+            await Next(context).ConfigureAwait(true);
+        }
+        catch (Exception ex) when (!context.Response.HasStarted)
+        {
+            (string ExceptionName, int StatusCode) exceptionInformation = ExceptionInformation(ex);
+
+            ResponseHeaderHandler.ClearResponse(context, TrackedExceptions.GetExceptionStatusCode(ex));
```

Comment:
Can't we just use status code we already have from the line above?

Reply:
Good catch! Will be fixed.